### PR TITLE
Backblaze Driver

### PR DESCRIPTION
**Pull request by Geometrically**
This PR adds a simple backblaze driver for support for uploading and deleting files to the Backblaze B2 Bucket.

It has tests for checking if the code is working correctly too.

### CODE REVIEW

File: src/file_hosting/authorization.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    let req = reqwest::Client::new()
+        .get("https://api.backblazeb2.com/b2api/v2/b2_authorize_account")
+        .header(reqwest::header::CONTENT_TYPE, "application/json")
+        .header(reqwest::header::AUTHORIZATION, formatted_key)
+        .send()
+        .await?;
+
+    let text = &req.text().await?;
+
+    Ok(serde_json::from_str(text)?)
```
Comment (Aeledfyr) [inline src/file_hosting/authorization.rs:49]: Reqwest has serde integration, so you can use `Ok(req.json()?)` instead if you add the `"json"` feature.  This is the same will all other instances of this, but I won't comment on them all.

Comment (Redblueflame) [reply src/file_hosting/authorization.rs:49]: +1

File: src/file_hosting/delete.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use crate::file_hosting::{AuthorizationData, FileHostingError};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct DeleteData {
```
Comment (Redblueflame) [inline src/file_hosting/delete.rs:7]: I think something like `DeleteFileData`, this is too generic

File: src/file_hosting/mod.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+#[derive(Error, Debug)]
+pub enum FileHostingError {
+    #[error("Error while accessing the data from remote")]
+    RemoteWebsiteError(#[from] reqwest::Error),
+    #[error("Error while serializing or deserializing JSON")]
+    SerDeError(#[from] serde_json::Error),
+}
+
+#[cfg(test)]
```
Comment (Redblueflame) [inline src/file_hosting/mod.rs:27]: The tests are including too much different functions, separing them into multiple ones and one final that test their integration together is needed

File: src/file_hosting/upload.rs [hunk 4]
```diff
@@ -0,0 +1,35 @@
+use serde::{Deserialize, Serialize};
+use crate::file_hosting::FileHostingError;
+use crate::file_hosting::authorization::UploadUrlData;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct UploadData {
```
Comment (Redblueflame) [inline src/file_hosting/upload.rs:7]: Same as before, `UploadFileData`

File: src/file_hosting/mod.rs (hunk 3 above)
Comment (Geometrically) [reply src/file_hosting/mod.rs:27]: the problem with that is all of the methods depend on each other so doing that is basically impossible

File: src/file_hosting/authorization.rs (hunk 1 above)
Comment (Geometrically) [reply src/file_hosting/authorization.rs:49]: This method doesn't exist in the async impl of reqwest.

Comment (Aeledfyr) [reply src/file_hosting/authorization.rs:48]: Is this not the correct method?  You have to enable the "json" feature, but it's there.  https://docs.rs/reqwest/0.10.6/reqwest/struct.Response.html#method.json

Comment (Geometrically) [reply src/file_hosting/authorization.rs]: Fixed.

**Pull request merged**